hr_holidays_half_day/models/hr_holidays.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Coop IT Easy SCRLfs
+#   - Vincent Van Rossem <vincent@coopiteasy.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import math
+from pytz import timezone, UTC
+
+from openerp import models, fields, api, _
+from openerp.exceptions import ValidationError
+
+
+def floatime_to_hour_minute(f):
+    decimal, integer = math.modf(f)
+    return int(integer), int(round(decimal * 60))
+
+
+class HrHolidays(models.Model):
+    _inherit = "hr.holidays"
+
+    period = fields.Selection(
+        string="Period",
+        selection=[("am", "AM"), ("pm", "PM"), ("day", "Day")],
+        default="day",
+    )
+
+    @api.onchange("holiday_status_id")
+    def _onchange_holiday_status_id(self):
+        self.number_of_days_temp = self._recompute_days()
+
+    def _recompute_days(self):
+        date_from = self.date_from
+        date_to = self.date_to
+        if (date_to and date_from) and (date_from <= date_to):
+            duration = self._compute_number_of_days(
+                self.employee_id.id, date_from, date_to
+            )
+            return duration
+
+    @api.multi
+    def onchange_employee(self, employee_id):
+        res = super(HrHolidays, self).onchange_employee(employee_id)
+        duration = self._recompute_days()
+        res["value"]["number_of_days_temp"] = duration
+        return res
+
+    @api.multi
+    def onchange_date_from(self, date_to, date_from):
+        res = super(HrHolidays, self).onchange_date_from(date_to, date_from)
+        employee_id = self.employee_id.id or self.env.context.get(
+            "employee_id", False
+        )
+        if (date_to and date_from) and (date_from <= date_to):
+            diff_day = self._compute_number_of_days(
+                employee_id, date_from, date_to
+            )
+            res["value"]["number_of_days_temp"] = diff_day
+        return res
+
+    @api.multi
+    def onchange_date_to(self, date_to, date_from):
+        res = super(HrHolidays, self).onchange_date_to(date_to, date_from)
+        employee_id = self.employee_id.id or self.env.context.get(
+            "employee_id", False
+        )
+        if (date_to and date_from) and (date_from <= date_to):
+            diff_day = self._compute_number_of_days(
+                employee_id, date_from, date_to
+            )
+            res["value"]["number_of_days_temp"] = diff_day
+        return res
+
+    @api.onchange("period")
+    def onchange_period(self):
+        if self.type != "add":
+            period = self.period
+            from_dt = fields.Datetime.from_string(self.date_from)
+            to_dt = fields.Datetime.from_string(self.date_to)
+
+            company = self.env["res.company"].browse(
+                self.employee_id.company_id.id
+            )
```

Comment:
company = self.employee_id.company_id do the same

Reply:
changed in new commit